x-pack/plugins/infra/public/components/logging/log_entry_flyout/log_entry_actions_menu.tsx

```diff
@@ -110,31 +110,32 @@ const getUptimeLink = (logItem: LogEntriesItem): LinkDescriptor | undefined => {
   };
 };
 
-const getAPMLink = (logItem: LogEntriesItem): LinkDescriptor | undefined => {
-  const traceIdEntry = logItem.fields.find(
-    ({ field, value }) => value[0] != null && field === 'trace.id'
-  );
+const getAPMLink = (logEntry: LogEntry): LinkDescriptor | undefined => {
+  const traceId = logEntry.fields.find(
+    ({ field, value }) => typeof value[0] === 'string' && field === 'trace.id'
+  )?.value?.[0];
 
-  if (!traceIdEntry) {
+  if (typeof traceId !== 'string') {
```

Comment:
Since we're not strigifying early on the server anymore we need to check more strictly. This could be any JSON-encodable value